### DISCUSSION TITLE
Allow us to use our Own GPT API key

### DISCUSSION THREAD

**Discussion by ahmadsyedanwar** [sweepai/sweep#1237] [Q&A]
Please add a feature that allows us to use our own OpenAI keys. Some of us have already acquired GPT-4 and want to use it for larger code solutions. This will help Sweep AI avoid high bills. @kevinlu1248

**Comment by kevinlu1248** [comment]
@ahmad-ml Most people don't have 32k access and there are security flaws we need to figure out around passing the secret to us. Do you have suggestions for this?

**Reply by ahmadsyedanwar** [reply]
https://www.npmjs.com/package/opencommit

We use this project as our Git commit helper. It takes the API key and manages it in our local environment file. You could try this approach.

**Reply by kevinlu1248** [reply]
I've used opencommit before but it runs locally and we run in the cloud so it's different. You can store your API key locally and opencommit will call that local key but Sweep doesn't get the same luxury. I can dig into how GitHub Action secrets work as a potential solution.

**Comment by FreePhoenix888** [comment]
As I know now there is a possibility to run self-hosted sweepai where you can use your own chatgpt. Should we close this discussion?

**Reply by kevinlu1248** [reply]
Yup closing now.
